### PR TITLE
Custom field and observation types

### PR DESCRIPTION
**Pull request by danjampro**
Adds functionality to use custom `Field` and `Observation` objects using the existing `Scheduler` paradigm. The existing types `panoptes.pocs.scheduler.field.Field` and `panoptes.pocs.scheduler.observation.base.Observation` are used by default, but can be overridden in the targets file.

This requires a format change to the targets file to add the `field` and `observation` keys. This seems like the best method of creating generic `Field` and `Observation` objects from a config file. 

Also this PR renames `fields_list` and `fields_file` to `targets_list` and `targets_file`, making it more consistent with the rest of `POCS` (e.g. `targets.yaml`, `directories.targets` etc).

This will pave the way for scheduling of things like calibs and other observations that do not work with the standard `Field/Observation`.

See: AstroHuntsman/huntsman-pocs#406

## How Has This Been Tested?
Unit tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by wtgee — CHANGES_REQUESTED** [review]
Thanks @danjampro. I'm a little torn on the renaming of everything to `targets` as I think that's a bit the opposite direction from how it should go; namely, everything should be changed toto `fields` where it currently says `target`, e.g. the directory should probably change to `directories.fields`, etc.

The `Observation` has a field, which also has a certain FOV, etc.  It could be said that it is a target field, but more often (at least for exoplanets), our target would probably be the star at the center of the field, which is why there is baked in confusion for POCS right now.

This would also keep it more consistent with the class name and what is in the config.

More practically speaking for requested changes, I don't see any tests of the new `type` field or example in the config.

File: src/panoptes/pocs/scheduler/scheduler.py [hunk 1]
```diff
@@ -209,31 +209,40 @@ def observation_available(self, observation, time):
...
+            target_config (dict): Configuration items for `Observation`
         """
-        with suppress(KeyError):
-            field_config['exptime'] = float(
-                get_quantity_value(field_config['exptime'], unit=u.second)) * u.second
+        target_config = target_config.copy()
+        self.logger.debug(f"Adding target_config={target_config!r} to scheduler.")
+
+        field_config = target_config.get("field", {})
+        field_type_name = field_config.pop("type", "panoptes.pocs.scheduler.field.Field")
```
[the remark this answers is not in the document]
Comment (danjampro) [reply src/panoptes/pocs/scheduler/scheduler.py]: Docstring added.

File: src/panoptes/pocs/scheduler/scheduler.py [hunk 2]
```diff
@@ -209,31 +209,40 @@ def observation_available(self, observation, time):
...
 
-        self.logger.debug(f"Adding field_config={field_config!r} to scheduler")
-        field = Field(field_config['name'], field_config['position'])
-        self.logger.debug(f"Created field.name={field.name!r}")
+        obs_config = target_config.get("observation", {})
+        obs_type_name = obs_config.pop(
+            "type", "panoptes.pocs.scheduler.observation.base.Observation")
```
[the remark this answers is not in the document]
Comment (danjampro) [reply src/panoptes/pocs/scheduler/scheduler.py]: Docstring added.

**Comment by danjampro** [comment]
> Thanks @danjampro. I'm a little torn on the renaming of everything to `targets` as I think that's a bit the opposite direction from how it should go; namely, everything should be changed toto `fields` where it currently says `target`, e.g. the directory should probably change to `directories.fields`, etc.
> 
> The `Observation` has a field, which also has a certain FOV, etc. It could be said that it is a target field, but more often (at least for exoplanets), our target would probably be the star at the center of the field, which is why there is baked in confusion for POCS right now.
> 
> This would also keep it more consistent with the class name and what is in the config.
> 
> More practically speaking for requested changes, I don't see any tests of the new `type` field or example in the config.

@wtgee thanks for the review! I have reverted to the old "fields" naming as well as renamed `directories.targets` to `directories.fields`. I have also updated the docstrings accordingly.

**Review by wtgee — APPROVED** [review]
👍🏼 thanks for addressing those changes and adding the new feature!

**Pull request merged**